src/streamlink/plugins/wasd.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.stream import HLSStream
+
+log = logging.getLogger(__name__)
+
+class WASD(Plugin):
+	_url_re = re.compile(r"https?://(?:www\.)?wasd\.tv/")
+	_url_channel = re.compile(r"https?://(?:www\.)?wasd\.tv/channel/(\d+)")
+	_url_video = re.compile(r"https?://(?:www\.)?wasd\.tv/channel/\d+/videos/(\d+)")
+	
+	@classmethod
+	def can_handle_url(cls, url):
+		return WASD._url_re.match(url)
+		
+	def _get_streams(self):
+		re_video_id = self._url_video.match(self.url)
+		re_channel_id = self._url_channel.match(self.url)
+		
+		if not re_video_id and not re_channel_id:
+			return log.error("Expected to see url like https://wasd.tv/channel/12345 or https://wasd.tv/channel/12345/videos/67890")
+		
+		# Check if we have AUTH token stored in session cache and request it
+		cookies = self.session.http.cookies
+		if not cookies or not cookies['cronos-auth-token'] or not cookies['cronos-auth-token-signature']:
+			log.debug("There is no AUTH token in cache. Attempting to request one")
+			self.getAuthToken()
```

Comment:
> self.getAuthToken()

just open the website here instead of `self.getAuthToken()` and let it fail here if you don't have both cookies

Reply:
Explain this please.
Cookies `cronos-auth-token` and `cronos-auth-token-signature` are required to make API request.
To get those cookies I need to request `https://wasd.tv/api/auth/anon-token` and it will response with `set-cookie` headers with life time of 7 days.
What you mean by "just open the website"?

Reply:
you are not reusing `self.getAuthToken` , so you don't need to create a new method for it

without it, it will be easier to read the code in the future in something breaks

---

you also don't need all the `if not res`
just use it with `res.cookies.get('XXX')`

---

some of the other methods are also not required